Jamulus.pro

```diff
@@ -1067,6 +1067,15 @@ contains(CONFIG, "headless") {
     FORMS += $$FORMS_GUI
 }
 
+contains(VERSION, .*dev.*) {
+    # Format all code via clang-format on non-release builds.
+    # If clang-format isn't available, we fail silently, but we check on PR submission.
+    # This must come before addition of opus sources:
+    clang_format.commands = 'clang-format -i $$SOURCES $$HEADERS || echo "clang-format-based source code formatting failed. Please ensure that clang-format is installed and available in your PATH variable."'
+    QMAKE_EXTRA_TARGETS += clang_format
+    PRE_TARGETDEPS += clang_format
```

Comment:
I would remove the `PRE_TARGETDEPS` line. Removing it would allow the `Makefile` to offer the ability to do `make clang_format` to easily update the format of files, but would not make it a dependency of `$(TARGET)` to force it do be done on every compilation.

For example, my RPi does have `clang-format`, but it is only version 7 and not compatible with the Jamulus `.clang-format`. I do a lot of development and compilation on it, and having `clang_format` as a dependency would be incredibly annoying.

Reply:
Thanks. I've removed the automatical build dep. As the whole block becomes optional then, there's no reason to reduce the `make clang_format` addition to dev builds.